src/network.c

```diff
@@ -93,6 +93,7 @@ static gboolean transfer(RaucTransfer *xfer, GError **error)
 	curl_easy_setopt(curl, CURLOPT_XFERINFODATA, xfer);
 	curl_easy_setopt(curl, CURLOPT_FAILONERROR, xfer);
 	curl_easy_setopt(curl, CURLOPT_ERRORBUFFER, errbuf);
+	curl_easy_setopt(curl, CURLOPT_MAXFILESIZE_LARGE, xfer->limit);
```

Comment:
Do we need to cast to `curl_off_t` here?

Reply:
If it was not `curl_off_t` before, we could do it for making it explicit. If it was `curl_off_t` before, than not ;)